presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -142,6 +143,13 @@ public HivePartitionResult getPartitions(ConnectorSession session, SemiTransacti
         return new HivePartitionResult(partitionColumns, partitions.build(), remainingTupleDomain, enforcedTupleDomain, hiveBucketHandle);
     }
 
+    private List<HiveBucket> getBucketList(Table table, TupleDomain<ColumnHandle> effectivePredicate)
+    {
+        return getHiveBucket(table, TupleDomain.extractFixedValues(effectivePredicate).get())
```

Comment:
Done. I was thinking this way, but then decided to have clear switch place between my code and old code. Separate two getters and logic that combines them. I especially like moving one more function into private.